### PR TITLE
[Builtins] Unscrew generators tests

### PR DESCRIPTION
**Pull request by effectfully**
Don't look here yet.

### CODE REVIEW

File: plutus-core/testlib/PlutusCore/Generators/Internal/Denotation.hs [hunk 1]
```diff
@@ -51,45 +54,54 @@ data DenotationContextMember term res =
 --   2. a bound variable of functional type with the result being @integer@
 --   3. the 'AddInteger' 'Builtin' or any other 'Builtin' which returns an @integer@.
 newtype DenotationContext term = DenotationContext
-    { unDenotationContext :: DMap (AsKnownType term) (Compose [] (DenotationContextMember term))
+    { unDenotationContext :: DMap TypeRep (Compose [] (DenotationContextMember term))
```
Comment (effectfully) [inline plutus-core/testlib/PlutusCore/Generators/Internal/Denotation.hs:57]: This is the gist of the PR: replacing `AsKnownType` with `TypeRep` everywhere. Allows us to get rid of that ridiculous `unsafeCoerce` that we had, support more builtins and refactoring builtins stuff becomes much easier, 'cause these tests no longer get in the way as much.

File: plutus-core/testlib/PlutusCore/Generators/Internal/Entity.hs [hunk 2]
```diff
@@ -111,28 +107,16 @@ revealUnique (Name name uniq) =
...
-    ]
-
--- | Generate a 'Term' along with the value it computes to,
--- having a generator of terms of built-in types.
-withCheckedTermGen
-    :: (uni ~ DefaultUni, fun ~ DefaultFun, Monad m)
-    => TypedBuiltinGenT (Plain Term uni fun) m
-    -> (forall a. AsKnownType (Plain Term uni fun) a ->
-            TermOf (Plain Term uni fun) (EvaluationResult (Plain Term uni fun)) ->
-                GenT m c)
```
Comment (effectfully) [inline plutus-core/testlib/PlutusCore/Generators/Internal/Entity.hs:129]: We no longer use this apparently. I'm curious what it was used for, but not too much for me to actually go and check.

File: plutus-core/plutus-ir/src/PlutusIR/Core/Plated.hs [hunk 3]
```diff
@@ -34,7 +34,7 @@ import PlutusCore.Name qualified as PLC
 import PlutusIR.Core.Type
 
 import Control.Lens hiding (Strict, (<.>))
-import Data.Functor.Apply
+import Data.Functor.Apply hiding ((<.*>))
```
[the remark this answers is not in the document]
Comment (effectfully) [reply plutus-core/plutus-ir/src/PlutusIR/Core/Plated.hs]: Oops, that's due to `cabal` doing different things within the nix shell and outside of it.

**Comment by effectfully** [comment]
Ready for review, but I guess nobody really cares.

**Review by michaelpj — APPROVED** [review]

File: plutus-core/plutus-core/src/PlutusCore/Builtin/TypeScheme.hs [hunk 4]
```diff
@@ -55,10 +57,10 @@ on readability of the Core output.
 -- E.g. @Text -> Bool -> Integer@ is encoded as @TypeScheme val [Text, Bool] Integer@.
 data TypeScheme val (args :: [GHC.Type]) res where
     TypeSchemeResult
-        :: (KnownTypeAst (UniOf val) res, MakeKnown val res)
+        :: (Typeable res, KnownTypeAst (UniOf val) res, MakeKnown val res)
         => TypeScheme val '[] res
     TypeSchemeArrow
-        :: (KnownTypeAst (UniOf val) arg, MakeKnown val arg, ReadKnown val arg)
+        :: (Typeable arg, KnownTypeAst (UniOf val) arg, MakeKnown val arg, ReadKnown val arg)
```
[the remark this answers is not in the document]
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/Builtin/TypeScheme.hs:63]: > I don't love this. 

It was your suggestion! I think.

Strictly speaking, `args` and `res` are visible to the outside, so we could factor the constraints out, but then they'll end up in `BuiltinMeaning` where `args` and `res` are no longer visible to the outside, and I think that would be significantly worse.

> Maybe a comment pointing out that `TypeScheme` isn't relevant for evaluation so it's okay to put lots of stuff in there?

Will do.

**Pull request merged**